core/src/main/scala/kafka/log/FileMessageSet.scala

```diff
@@ -184,44 +185,44 @@ class FileMessageSet private[kafka](@volatile var file: File,
     new IteratorTemplate[MessageAndOffset] {
       var location = start
       val sizeOffsetBuffer = ByteBuffer.allocate(12)
-      
+
       override def makeNext(): MessageAndOffset = {
         if(location >= end)
           return allDone()
-          
+
         // read the size of the item
         sizeOffsetBuffer.rewind()
         channel.read(sizeOffsetBuffer, location)
         if(sizeOffsetBuffer.hasRemaining)
           return allDone()
-        
+
         sizeOffsetBuffer.rewind()
         val offset = sizeOffsetBuffer.getLong()
         val size = sizeOffsetBuffer.getInt()
         if(size < Message.MinHeaderSize)
           return allDone()
         if(size > maxMessageSize)
-          throw new InvalidMessageException("Message size exceeds the largest allowable message size (%d).".format(maxMessageSize))
-        
+          throw new CorruptRecordException("Message size exceeds the largest allowable message size (%d).".format(maxMessageSize))
```

Comment:
All server side exceptions should be the new o.a.k exceptions like `CorruptRecordException` and should be handled correctly by `Errors.java`. With this patch the only place `ErrorMapping` is used is in client code. 

We throw `InvalidMessageException` from `kafka.javaapi.message.MessageSet` on the client side, not from `kafka.message.MessageSet` which is used both by client and server.